examples/seismic/viscoelastic/wavesolver.py

```diff
@@ -0,0 +1,133 @@
+from devito.tools import memoized_meth
+from examples.seismic import Receiver
+from examples.seismic.viscoelastic.operators import (ForwardOperator, tensor_function,
+                                                     vector_function)
+
+
+class ViscoelasticWaveSolver(object):
+    """
+    Solver object that provides operators for seismic inversion problems
+    and encapsulates the time and space discretization for a given problem
+    setup.
+
+    Parameters
+    ----------
+    model : Model
+        Physical model with domain parameters.
+    geometry : AcquisitionGeometry
+        Geometry object that contains the source (SparseTimeFunction) and
+        receivers (SparseTimeFunction) and their position.
+    space_order : int, optional
+        Order of the spatial stencil discretisation. Defaults to 4.
+
+    Notes
+    -----
+    This is an experimental staggered grid viscoelastic modeling kernel.
+    """
+    def __init__(self, model, geometry, space_order=4, **kwargs):
+        self.model = model
+        self.geometry = geometry
+
+        self.space_order = space_order
+        # Time step can be \sqrt{3}=1.73 bigger with 4th order
```

Comment:
Maybe revisit this relation, `bigger than what`?

Reply:
Comment removed for now.